lib/daru/view/adapters/highcharts/display.rb

```diff
@@ -33,13 +33,25 @@ class HighChart
     #
     def to_html(placeholder=random_canvas_id)
       chart_hash_must_be_present
-      high_chart(placeholder, self)
+      # Helps to denote either of the three classes.
+      chart_class = extract_chart_class
+      # When user wants to plot a HighMap
+      if chart_class == 'Map'
+        high_map(placeholder, self)
+      # When user wants to plot a HighStock
+      elsif chart_class == 'StockChart'
+        high_stock(placeholder, self)
+      # No need to pass any value for HighChart
+      elsif chart_class == 'Chart'
+        high_chart(placeholder, self)
+      else
+        raise 'chart_class must be selected as either chart, stock or map'
```

Comment:
Isn't this `raise` must be in `extract_chart_class` method?  Remember one method must do only one thing.